# @package _global_
use_cuda: True
cuda_device_num: 0

control_dt: 0.015
#dynamics_dt: 0.01
run_open_loop: False
control_space: 'acc'
float_dtype: 'float32'
model:
  # any link that is not specified as learnable will be initialized from urdf
  urdf_path: "urdf/franka_description/franka_panda_no_gripper.urdf"
  learnable_rigid_body_config:
    learnable_links: []
  name: "franka_panda"
  dt: 0.008
  max_acc: 1.0 #10.0

  dt_traj_params:
    base_dt: 0.008
    base_ratio: 1.0
    max_dt: 0.1
  #control_freq: 10
  #sim_freq: 100
  ee_link_name: "ee_link"
  init_state: [0.8, 0.3, 0.0, -1.57, 0.0, 1.86, 0.]
  link_names: ['panda_link2','panda_link3', 'panda_link4', 'panda_link5', 'panda_link6','panda_link7','panda_hand']

  collision_model:
    link_objs:
      panda_hand:
        base: [0.0, 0.0, 0.0]
        tip: [0.0, 0.05, 0.0]
        radius: 0.06
        pose_offset: [0.0, 0.0, 0.12, 0.0, 0.0, 0.0] # x,y,z, r,p,y


cost:
  goal_pose:
    vec_weight: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
    weight: [50.0,100.0]
    position_gaussian_params: {'n':0, 'c':0.2, 's':0.0, 'r':10.0}
    orientation_gaussian_params: {'n':0, 'c':0.2, 's':0.0, 'r':10.0}
    hinge_val: -1
    convergence_val: [0.0, 0.0] # orientation, position

  zero_vel:
    weight: 0.0
    hinge_val: 0.1
    gaussian_params: {'n':0, 'c':0.1, 's':0, 'r':1.0}

  zero_acc:
    weight: 0.0 #0.01
    hinge_val: -1.0
    gaussian_params: {'n':0, 'c':0.2, 's':0, 'r':5.0}
  
  collision:
    weight: 0.0
    gaussian_params: {'n':0, 'c':0.5, 's':0, 'r':5.0}
    
  primitive_collision:
    weight: 0.0
    gaussian_params: {'n':0, 'c':0.8, 's':0, 'r':10.0}

  null_space:
    weight: 0.1 #0.05 #0.5 #1.0
    gaussian_params: {'n':0, 'c':0.5, 's':0, 'r':1.0}
    vec_weight: [0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0]
    #retract_state : [-1.57, -1.2, 1.9, -1.383, -1.57, 0.00]
  jacobian_cost_weight: 0.0 #2

  joint_l2:
    weight: 0.0
    gaussian_params: {'n':0, 'c':0.2, 's':0, 'r':5.0}

  ee_vel:
    weight: 1.0
    vec_weight: [1.0, 1.0, 1.0, 0.0, 0.0, 0.0]
    gaussian_params: {'n':0, 'c':0.2, 's':0.0, 'r':5.0}
    
  scene_collision:
    weight: 5000.0 #5000.0
    gaussian_params: {'n':0, 'c':0.8, 's':0, 'r':10.0}

  control_cost_weight: 0.0
  jerk_cost_weight: 0.0
  
  retract_state : [0.8, 0.3, 0.0, -1.57, 0.0, 1.86, 0.]
  retract_weight: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0,0.0]

mppi:
  horizon           : 50
  init_cov          : 0.001 #0.5
  gamma             : 0.99
  n_iters           : 1
  step_size_mean    : 0.9
  step_size_cov     : 0.9
  beta              : 1.0
  alpha             : 1
  num_particles     : 400 #10000
  update_cov        : False
  cov_type          : 'diag_AxA' # 
  kappa             : 0.1
  null_act_frac     : 0.01
  sample_mode       : 'mean'
  base_action       : 'repeat'
  squash_fn         : 'clamp'
  hotstart          : True  
  sample_params:
    type: 'halton_stomp'
    fixed_samples: True
    sample_ratio: 0.5
    seed: 0
    filter_coeffs: [0.2, 0.3, 0.5]
